### PR TITLE
Preliminary work on encryption

### PR DESCRIPTION
**Pull request by jbaublitz**
Initial review only.

### CODE REVIEW

File: src/dbus_api/pool.rs [hunk 1]
```diff
@@ -281,6 +281,81 @@ fn add_cachedevs(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
     add_blockdevs(m, BlockDevTier::Cache)
 }
 
+fn add_cachedevs_2_1(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
```
Comment (jbaublitz) [inline src/dbus_api/pool.rs]: This method is broken.

File: tests/client-dbus/src/stratisd_client_dbus/_data.py [hunk 2]
```diff
@@ -59,6 +59,33 @@
 <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="const"/>
 </property>
 </interface>
+""",
```
Comment (jbaublitz) [inline tests/client-dbus/src/stratisd_client_dbus/_data.py]: Add to this XML file to reflect the actual interface.

**Comment by bgurney-rh** [comment]
I'm not sure if this is something caused by this PR, but I saw the stratisd test hang last time, at about this stage:
```
test engine::strat_engine::pool::tests::real_test_add_cachedevs
```

I had to cancel the test run, and there was a thin pool that was suspended.  On another test run, I tried to resume the thin pool after seeing it hung for 30 minutes; it caused the test above to fail, but the rest of the test run continued (it's currently running).

**Comment by bgurney-rh** [comment]
I caught another test that seemed to be holding a thin pool suspended for a while:
```
test engine::strat_engine::thinpool::thinpool::tests::real_test_set_device ... FAILED
```

This time, the stratisd test failed (which is better than hanging for 5 days.  Here's a quick summary of the tests that failed:
```
test engine::strat_engine::backstore::backstore::tests::real_test_add_cache_devs ... FAILED
test engine::strat_engine::backstore::backstore::tests::real_test_setup ... FAILED
test engine::strat_engine::backstore::blockdevmgr::tests::real_test_blockdevmgr_used ... FAILED
test engine::strat_engine::pool::tests::real_test_add_cachedevs ... FAILED
test engine::strat_engine::thinpool::thinpool::tests::real_test_set_device ... FAILED
```

I was on the test system, and if I saw it hanging for about 20 minutes or so, with a suspended thin pool, I resumed the thin pool.  I believe the only times I did that were for the two tests noted above.

File: src/dbus_api/pool.rs [hunk 3]
```diff
@@ -219,6 +219,61 @@ fn snapshot_filesystem(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
     Ok(vec![msg])
 }
 
+fn init_cache_2_1(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
```
Comment (mulkieran) [inline src/dbus_api/pool.rs]: There seems to be so much code shared between ```add_blockdevs``` and this method that there is probably a lot of room for abstraction and unification between the two.

**Comment by bgurney-rh** [comment]
I'm still getting a stuck test system due to this, but I found something else: an open test filesystem in the path "/tmp/stratis_testing6Dw2e1".

I can reproduce this on a test vm running the "test_real" test.  I can see if I can retrofit the teardown.sh script to clean up after it, but I haven't had to do this until now, which suggests that there may be something leaking out of test_add_cachedevs() with the current changes in this PR.

**Comment by mulkieran** [comment]
@jbaublitz I may not have gotten to the root cause of the error, but you will have to suspend the pool in ```init_cache```.

**Comment by bgurney-rh** [comment]
I had one last hang on "test engine::strat_engine::thinpool::thinpool::tests::real_test_set_device" when I tried it; looks like the same thing.

File: src/dbus_api/macros.rs [hunk 4]
```diff
@@ -30,6 +30,19 @@ macro_rules! get_parent {
     };
 }
 
+/// Macro for early return with Ok dbus message on failure to get mutable pool.
```
[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/dbus_api/macros.rs]: Resolved.

File: src/dbus_api/pool.rs (hunk 3 above)
Comment (jbaublitz) [reply src/dbus_api/pool.rs]: Resolved.

File: tests/client-dbus/src/stratisd_client_dbus/_data.py (hunk 2 above)
Comment (jbaublitz) [reply tests/client-dbus/src/stratisd_client_dbus/_data.py]: Resolved.

**Comment by mulkieran** [comment]
@jbaublitz Please rebase, conflict recently introduced.

**Comment by jbaublitz** [comment]
@mulkieran Resolved.

**Pull request closed without merging**